### PR TITLE
Stocks

### PR DESCRIPTION
**Pull request by mschug**
Quandl API is being reached in api.js, and is returning the correct information for 4 out of 5 of the given stocks. DJI is an index and therefore not in the same database as the other stocks, so we'll need to add a special case for when a stock isn't found.

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -23,6 +23,49 @@ var events = [
...
+  {
+    symbol:'XRX',
+    price:'9.21',
+    high:'9.34',
+    low:'8.93',
+    yrhigh:'12.25',
+    yrlow:'8.32'
+  }
+];
+
```
Comment (JRJurman) [inline api/index.js:68]: remove this if the stocks/:symbol endpoint works

File: api/index.js [hunk 2]
```diff
@@ -40,5 +83,25 @@ api.get('/events', (req, res) => {
   res.send(events);
 });
 
+api.get('/stocks/:symbol', (req, res) => {
+  var Quandl = require('quandl');
```
Comment (bbesmanoff) [inline api/index.js:87]: `npm install --save-dev quandl`

File: api/index.js [hunk 3]
```diff
@@ -40,5 +83,25 @@ api.get('/events', (req, res) => {
...
+  var quandl = new Quandl({
+    auth_token: process.env.quandl,
```
Comment (bbesmanoff) [inline api/index.js:89]: is dotenv configured for this yet?

File: app/js/components/ticker-list.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    super(props);
+    this.state = {stocks:[]};
+  }
+
+  componentDidMount () {
+    var stockRequest = new XMLHttpRequest();
+    var symbol = this.props.symbol;
+    var stockInfo, stocks;
+    if (this.props.symbol) {
+      stockRequest.open('GET', (`./api/stocks/${symbol}`));
```
Comment (bbesmanoff) [inline app/js/components/ticker-list.js:16]: `'GET', '/api/stocks/${symbol}'`

File: api/index.js (hunk 2 above)
Comment (JRJurman) [inline api/index.js:87]: this should be at the top of the file?

File: api/index.js (hunk 3 above)
Comment (mschug) [inline api/index.js:89]: I thought it had been. So far I've been hard-coding the API key for testing and replacing it with the dotenv before making a commit.

**Comment by JRJurman** [comment]
can we get a screenshot for the description? what does this visually look like?

File: app/js/index.js [hunk 5]
```diff
@@ -17,6 +22,11 @@ class App extends React.Component {
           <div className="row">
             <div className="col-xs-4">
               <h1>Stocks</h1>
+              <TickerList symbol="AAPL" />
+              <TickerList symbol="DJI" />
+              <TickerList symbol="GOOG" />
+              <TickerList symbol="XOM" />
+              <TickerList symbol="XRX" />
```
Comment (JRJurman) [inline app/js/index.js]: why are there several items here?

File: app/js/components/ticker-list.js (hunk 4 above)
Comment (bbesmanoff) [inline app/js/components/ticker-list.js:16]: Also, this seems like it's a little misplaced - `TickerList` implies that this is a list and not an individual Ticker

File: app/js/index.js [hunk 6]
```diff
@@ -2,7 +2,12 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 
 import Navbar from './components/navbar';
+import Event from './components/event';
+import Ticker from './components/ticker';
+
+//const {Route, RouteHandler} = Router;
```
Comment (bbesmanoff) [inline app/js/index.js]: Rebase against master - this was deleted, but is now an addition of `//const {Route, RouteHandler} = Router;`

File: app/js/components/ticker-list.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    else {
+      stockRequest.open('GET', './api/stocks');
```
Comment (JRJurman) [inline app/js/components/ticker-list.js]: change endpoint name?
possibly top-stocks, or favorite-stocks

**Comment by JRJurman** [comment]
can we remove this PR? I think everything in here has been merged in by now.

**Comment by bbesmanoff** [comment]
I think it can be removed

**Pull request closed without merging**